### PR TITLE
[CR] [WIP] Shields

### PR DESCRIPTION
**Pull request by ThinkInvis**
JSON:
- [x] Items in JSONs.
- [ ] Check prices, volumes, weights, etc.
- [ ] Recipes (and books for recipes) in JSONs.
- [ ] Added to item groups if applicable. (Military outposts? Barricades? Museums?)

Code:
- [x] Basic bonus to block if equipped.
- [ ] Move individual item properties to JSONs, if possible; make sure item properties in JSON aren't actually contributing to armor values since these are technically 'off-hand weapons'
- [ ] Prevent equipping multiple shields at once. (?)
- [ ] Allow larger shields to be planted on the ground ("ENTRENCH" flag?)
- [ ] 'Shield stance' - hide behind your shield - more bonus to block and more encumbrance while active
- [ ] Shield bashing (?)
- [ ] Enemies can disarm or pull down shields (?)

Misc. todo and discussion:
- (?) Not sure whether this is a good idea.
- Should the 'unable to block projectiles', 'bulletproof', etc. concepts on some of the shields actually be applied; if not, should I remove that specific flavortext?
- As stated above - if an item stat doesn't make sense (especially the block values for shields, which i pulled out of nowhere) please suggest a better one. I'm not very familiar with the numbers behind some of these stats yet.
- Was limiting block_score to 40 actually necessary or is there a check within the block logic that doesn't allow the player to be healed or some such nonsense by blocking more than 1.0?
- Should the atlatl actually make a faint 'zip' sound? According to THE INTERNET, they do, but I haven't seen one since a school field trip more than a decade ago so I wouldn't know.

An attempt at adding shields and atlatls to C:DDA.   The poor WBLOCK_3 tag is still unused :(

### CODE REVIEW

**Comment by Lain-** [comment]
Maybe not in the scope of this PR, but I think it's important to consider that if shields are going in, we should think about introducing distinction between one-handed and two-handed weapon too. Along with all the questions that entail like "what will happen when shield + two-handed (staff, longsword)?", "using one-handed weapons with both hands?", "dual-wield?".

**Comment by kevingranade** [comment]
Atlatls and shields are two major changes that have nothing to do with each
other. I strongly suggest you split their development into two PRs.  I'd
suggest atlatls first as they are probably far simpler.

**Comment by ThinkInvis** [comment]
I just had two ideas at once and didn't want to lose track of one of them, and I'm not too familiar with git. How would I remove the changes from this PR as well as the branch and add them to a new branch?

**Comment by Snaaty** [comment]
Just leaving a comment here for future updates and for encouragement. Keep it up!

**Comment by BevapDin** [comment]
`WBLOCK_[1-3] ` have been changed to be techniques, they are checked in melee.cpp and should affect the dodging. At least several items in melee.json and tools.json have those techniques, but `WBLOCK_3` is indeed not assigned to any real item.

**Comment by kevingranade** [comment]
Since you added them in the same commit, there's not much for it other than to copy some of the changes to a new branch and manually revert them from this branch.

File: data/json/items/ranged.json [hunk 1]
```diff
@@ -4006,7 +4006,7 @@
     "to_hit": -1,
     "ranged_damage": 1,
     "range": 45,
-    "dispersion": 30,
+    "dispersion": 150,
```
Comment (kevingranade) [inline data/json/items/ranged.json]: Looks like you accidentally reverted a recent change to the browning.

Comment (ThinkInvis) [inline data/json/items/ranged.json]: Yeah, I knew about that. Just git being silly.

**Comment by ThinkInvis** [comment]
Splitting atlatls into a second PR. Managed to seriously mess up the throw_range function's formatting while removing code :(

**Comment by Toybasher** [comment]
I like it. What if the shield's had even more statistics? Not sure the capabilities of the game engine but...

For example, a plexiglass shield should be able to stop a few low calibur bullets before shattering, while a metal one can take much more punishment. (An anti material turret should still punch right through and seriously wound/kill you though.) A plastic riot shield should stop most melee attacks but something sharp or a bullet would go right through it. What about the ability to affix a sharp spike onto the shield to be used for impaling zombies? Or the chance for a melee attack with fists to hit the spike, dealing damage to the attacker?

**Comment by chaosvolt** [comment]
I like the idea behind this. The quick-and-dirty way to allow shield block, by giving coverage to hands, arms, and torso would be a simple way to handle it, though the idea of it granting a block ability seems more clever than the ideas for shields I've dealt with in the past.

Having coverage and encumbrance affect things would be interesting as well, but those are perhaps best for the "shields as raw armor" take on the concept.

I might in fact suggest giving blocking or improved blocking to the most easy-to-manipulate shields, like the basic round shield, heater shield, buckler, etc. Larger shields might instead warrant NOT granting blocking, and instead protecting the player solely based of body parts affected and its listed coverage.

**Comment by OzoneH3** [comment]
No progress for nearly a year, closing.

**Comment by chaosvolt** [comment]
Well, there's always still the hackish shields I've been saving in the full version of medieval mod. ;w;

**Pull request closed without merging**